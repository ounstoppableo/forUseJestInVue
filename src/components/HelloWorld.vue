<script setup>
import { ref } from 'vue'
import {getList} from '../api/api'
const tableData = ref([])
getList().then(res=>{
  tableData.value = res.data
})
</script>

<template>
  <el-table :data="tableData" style="width: 100%">
    <el-table-column prop="date" label="Date" width="180" />
    <el-table-column prop="name" label="Name" width="180" />
    <el-table-column prop="address" label="Address" />
  </el-table>
</template>

<style scoped>

</style>
