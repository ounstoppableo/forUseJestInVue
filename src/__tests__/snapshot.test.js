import HelloWorld from '../components/HelloWorld.vue';
import { mount } from '@vue/test-utils';
import {listData} from './__mocks__/mock';
import flushPromises from 'flush-promises';
import {getList} from '../api/api';
describe('快照最佳实践', () => {
    test('对HelloWorld.vue文件的快照测试', async () => {
        getList.mockReturnValue(Promise.resolve(listData));
        const wrapper = mount(HelloWorld);
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    })
})