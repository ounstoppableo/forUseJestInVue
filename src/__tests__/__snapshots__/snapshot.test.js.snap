// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照最佳实践 对HelloWorld.vue文件的快照测试 1`] = `
"<div class="el-table--fit el-table--enable-row-hover el-table--enable-row-transition el-table el-table--layout-fixed" style="width: 100%;" data-prefix="el">
  <div class="el-table__inner-wrapper">
    <div class="hidden-columns">
      <div></div>
      <div></div>
      <div></div>
    </div>
    <div class="el-table__header-wrapper">
      <table class="el-table__header" border="0" cellpadding="0" cellspacing="0">
        <colgroup>
          <col name="el-table_1_column_1" width="180">
          <col name="el-table_1_column_2" width="180">
          <col name="el-table_1_column_3" width="80">
        </colgroup>
        <thead class="">
          <tr class="">
            <th class="el-table_1_column_1 is-leaf el-table__cell" colspan="1" rowspan="1">
              <div class="cell">Date
                <!---->
                <!---->
              </div>
            </th>
            <th class="el-table_1_column_2 is-leaf el-table__cell" colspan="1" rowspan="1">
              <div class="cell">Name
                <!---->
                <!---->
              </div>
            </th>
            <th class="el-table_1_column_3 is-leaf el-table__cell" colspan="1" rowspan="1">
              <div class="cell">Address
                <!---->
                <!---->
              </div>
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div class="el-table__body-wrapper">
      <div class="el-scrollbar">
        <div class="el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
          <div class="el-scrollbar__view" style="display: inline-block; vertical-align: middle;">
            <table class="el-table__body" cellspacing="0" cellpadding="0" border="0" style="table-layout: fixed;">
              <colgroup>
                <col name="el-table_1_column_1" width="180">
                <col name="el-table_1_column_2" width="180">
                <col name="el-table_1_column_3" width="80">
              </colgroup>
              <!--v-if-->
              <tbody tabindex="-1">
                <tr class="el-table__row">
                  <td class="el-table_1_column_1 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->2016-05-03
                    </div>
                  </td>
                  <td class="el-table_1_column_2 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->Tom
                    </div>
                  </td>
                  <td class="el-table_1_column_3 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->No. 189, Grove St, Los Angeles
                    </div>
                  </td>
                </tr>
                <tr class="el-table__row">
                  <td class="el-table_1_column_1 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->2016-05-02
                    </div>
                  </td>
                  <td class="el-table_1_column_2 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->Tom
                    </div>
                  </td>
                  <td class="el-table_1_column_3 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->No. 189, Grove St, Los Angeles
                    </div>
                  </td>
                </tr>
                <tr class="el-table__row">
                  <td class="el-table_1_column_1 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->2016-05-04
                    </div>
                  </td>
                  <td class="el-table_1_column_2 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->Tom
                    </div>
                  </td>
                  <td class="el-table_1_column_3 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->No. 189, Grove St, Los Angeles
                    </div>
                  </td>
                </tr>
                <tr class="el-table__row">
                  <td class="el-table_1_column_1 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->2016-05-01
                    </div>
                  </td>
                  <td class="el-table_1_column_2 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->Tom
                    </div>
                  </td>
                  <td class="el-table_1_column_3 el-table__cell" rowspan="1" colspan="1">
                    <div class="cell">
                      <!---->No. 189, Grove St, Los Angeles
                    </div>
                  </td>
                </tr>
              </tbody>
              <!--v-if-->
            </table>
            <!--v-if-->
            <!--v-if-->
          </div>
        </div>
        <transition-stub name="el-scrollbar-fade" appear="false" persisted="true" css="true">
          <div class="el-scrollbar__bar is-horizontal" style="display: none;">
            <div class="el-scrollbar__thumb" style="transform: translateX(0%);"></div>
          </div>
        </transition-stub>
        <transition-stub name="el-scrollbar-fade" appear="false" persisted="true" css="true">
          <div class="el-scrollbar__bar is-vertical" style="display: none;">
            <div class="el-scrollbar__thumb" style="transform: translateY(0%);"></div>
          </div>
        </transition-stub>
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="el-table__column-resize-proxy" style="display: none;"></div>
</div>"
`;
